pkg/observable/channel/map.go

```diff
@@ -112,6 +121,7 @@ func goMapTransformNotification[S, D any](
 	srcObserver observable.Observer[S],
 	transformFn MapFn[S, D],
 	publishFn func(dstNotifications D),
+	doneFn func(),
```

Comment:
#PUC on what's expected of this function.

Reply:
Could this be a channel instead of an anonymous function? Then we would just call `close` on 134 instead of the func instead.

I think the comment should be something to the effect of: it's the sender's responsibility to close the channel, since the publish channel (`dstProducer`) and its associated observable are constructed in this scope, if they're not also closed here then the associated goroutine will leak.

Reply:
Could this be a channel instead of an anonymous function? Then we would just call `close` on 134 instead of the func instead.

I think the comment should be something to the effect of: it's the sender's responsibility to close the channel, since the publish channel (`dstProducer`) and its associated observable are constructed in this scope, if they're not also closed here then the associated goroutine will leak. Closing the channel ends the goroutine.